src/main/java/org/jdata/timeseries/processing/Rule.java

```diff
@@ -0,0 +1,97 @@
+package org.jdata.timeseries.processing;
+
+/**
+ * @author Hunter
+ *
+ */
+public class Rule {
+
+    private int id;
+    private int refCount;
+    private Symbol head;
+
+    public Rule(int id) {
+        this.id = id;
+        refCount = 0;
+        head = new Symbol(0);
+        head.setNext(head);                 
+        head.setPrev(head);
+    }
+
+    public void addReference() {
+        this.refCount = refCount + 1;
+    }
+
+    public void removeReference() {
+        this.refCount = refCount - 1;
+    }
+
+    public void append(Symbol last) {
+        last = head.getPrev();
```

Comment:
This append method is causing tests to fail. Draw a diagram of what you want this method to do and try to get the tests to pass.